crates/accelerate/src/permutation.rs

```diff
@@ -77,6 +77,65 @@ fn get_ordered_swap(pattern: &ArrayView1<i64>) -> Vec<(i64, i64)> {
     swaps
 }
 
+fn pattern_to_cycles(pattern: &ArrayView1<i64>, invert_order: &bool) -> Vec<Vec<usize>> {
+    // vector keeping track of which elements in the permutation pattern have been visited
+    let mut explored: Vec<bool> = vec![false; pattern.len()];
+
+    // vector to store the cycles
+    let mut cycles: Vec<Vec<usize>> = Vec::new();
+
+    // cast the input pattern in terms of integers to usize, such that it can be used as index
+    // also invert the bit ordering if ``invert_order`` is true
+    let permutation: Array1<usize> = if *invert_order {
+        invert(pattern) // implies cast to usize
+    } else {
+        // TODO change to using Array1 insted of Vec
```

Comment:
Does this TODO refer to the output type? (returning Array1?), because it seems like pattern is already an Array1.

Reply:
Whoops, forgot to remove the comment after fixing it. Thanks for noticing!